#! /bin/sh -e
## 02_soname_map.dpatch by Eduard Bloch <blade@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix for the hard-coded DLL map

if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p1 ${patch_opts} < $0;;
    -unpatch) patch -R -p1 ${patch_opts} < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0

@DPATCH@
diff -urNad /home/inet/debian/dev/build-area/mono-0.31/data/config.in mono-0.31/data/config.in
--- /home/inet/debian/dev/build-area/mono-0.31/data/config.in	2004-04-07 00:18:29.000000000 +0200
+++ mono-0.31/data/config.in	2004-04-07 00:21:56.000000000 +0200
@@ -4,18 +4,26 @@
 	<dllmap dll="libxslt.dll" target="libxslt.so" />
 	<dllmap dll="libmySQL.dll" target="libmysqlclient.so" />
 	<dllmap dll="odbc32.dll" target="libodbc.so" />
-	<dllmap dll="libglib-2.0-0.dll" target="libglib-2.0.so" />
-	<dllmap dll="libgobject-2.0-0.dll" target="libgobject-2.0.so" />
-	<dllmap dll="libgthread-2.0-0.dll" target="libgthread-2.0.so" />
-	<dllmap dll="libpango-1.0-0.dll" target="libpango-1.0.so" />
-	<dllmap dll="libatk-1.0-0.dll" target="libatk-1.0.so" />
-	<dllmap dll="libgdk-win32-2.0-0.dll" target="libgdk-x11-2.0.so" />
-	<dllmap dll="libgdk_pixbuf-2.0-0.dll" target="libgdk_pixbuf-2.0.so" />
-	<dllmap dll="libgtk-win32-2.0-0.dll" target="libgtk-x11-2.0.so" />
-	<dllmap dll="libgtk-win32-2.0-0.dll" target="libgtk-x11-2.0.so" />
-	<dllmap dll="libglade-2.0-0.dll" target="libglade-2.0.so" />
+	<dllmap dll="libglib-2.0-0.dll" target="libglib-2.0.so.0" />
+	<dllmap dll="libgobject-2.0-0.dll" target="libgobject-2.0.so.0" />
+	<dllmap dll="libgthread-2.0-0.dll" target="libgthread-2.0.so.0" />
+	<dllmap dll="libpango-1.0-0.dll" target="libpango-1.0.so.0" />
+	<dllmap dll="libatk-1.0-0.dll" target="libatk-1.0.so.0" />
+	<dllmap dll="libgdk-win32-2.0-0.dll" target="libgdk-x11-2.0.so.0" />
+	<dllmap dll="libgdk_pixbuf-2.0-0.dll" target="libgdk_pixbuf-2.0.so.0" />
+	<dllmap dll="libgtk-win32-2.0-0.dll" target="libgtk-x11-2.0.so.0" />
+	<dllmap dll="libgtk-win32-2.0-0.dll" target="libgtk-x11-2.0.so.0" />
+	<dllmap dll="libglade-2.0-0.dll" target="libglade-2.0.so.0" />
 	<dllmap dll="oci" target="clntsh" />
 	<dllmap dll="gtkhtml-3.0" target="libgtkhtml-3.0.so" />
 	<dllmap dll="db2cli" target="libdb2_36.so"/>
-	<dllmap dll="gstreamer" target="libgstreamer-0.6.so" />
+  <dllmap dll="gnome-2" target="libgnome-2.so.0" />
+  <dllmap dll="gconf-2" target="libgconf-2.so.4" />
+  <dllmap dll="gstreamer" target="libgstreamer-0.6.so.0" />
+  <dllmap dll="gtksourceview-1.0" target="libgtksourceview-1.0.so.0">
+  <dllmap dll="gnomeui-2" target="libgnomeui-2.so.0" />
+  <dllmap dll="gobject-2" target="libgobject-2.0.so.0" />
+  <dllmap dll="gobject-2.0" target="libgobject-2.0.so.0" />
+  <dllmap dll="gnomevfs-2" target="libgnomevfs-2.so.0" />
+  <dllmap dll="gnomecanvas-2" target="libgnomecanvas-2.so.0" />
 </configuration>
